### PR TITLE
[Android] Add highlight support to inline text runs

### PR DESCRIPTION
**Pull request by almedina-ms**
Adds support for highlight in inline text runs actually completing this issue: https://github.com/Microsoft/AdaptiveCards/issues/2380

### CODE REVIEW

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/TextRendererUtil.java [hunk 1]
```diff
@@ -65,6 +65,12 @@ static String getTextColor(ForegroundColor foregroundColor, HostConfig hostConfi
         return hostConfig.GetForegroundColor(containerStyle, foregroundColor, isSubtle);
     }
 
+    static String getHighlightColor(ForegroundColor foregroundColor, HostConfig hostConfig, boolean isSubtle, ContainerStyle containerStyle)
+    {
+
```
Comment (paulcam206) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/TextRendererUtil.java]: nit: don't need the extra newline here

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/RichTextBlockRenderer.java [hunk 2]
```diff
@@ -107,6 +108,13 @@ else if ((textRun = TextRun.dynamic_cast(inline)) == null)
 
                 int color = getColor(TextRendererUtil.getTextColor(textRun.GetTextColor(), hostConfig, textRun.GetIsSubtle(), renderArgs.getContainerStyle()));
                 paragraph.setSpan(new ForegroundColorSpan(color), spanStart, spanEnd, Spannable.SPAN_INCLUSIVE_EXCLUSIVE);
+
+                if(textRun.GetHighlight())
```
Comment (paulcam206) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/RichTextBlockRenderer.java]: >f( [](start = 17, length = 2)

nit: spacing

File: source/shared/cpp/ObjectModel/ParseUtil.cpp [hunk 3]
```diff
@@ -54,6 +54,20 @@ namespace AdaptiveSharedNamespace
...
+        std::string propertyName = AdaptiveCardSchemaKeyToString(key);
+        auto propertyValue = json.get(propertyName, Json::Value());
+        if (propertyValue.empty() || !propertyValue.isString())
+        {
+            return "";
+        }
+        else
+        {
+            return propertyValue.asString();
+        }
```
Comment (paulcam206) [inline source/shared/cpp/ObjectModel/ParseUtil.cpp]: isn't this just `ParseUtil::GetString(json, key, "", false);`?

Comment (RebeccaAnne) [reply source/shared/cpp/ObjectModel/ParseUtil.cpp]: ParseUtil::GetString(json, key, "", false); will throw if the value at key exists but is not a string. In the case this is handling, the value at key may very well be an object.

---
In reply to: [271454369](https://github.com/Microsoft/AdaptiveCards/pull/2609#discussion_r271454369) [](ancestors = 271454369)

Comment (RebeccaAnne) [reply source/shared/cpp/ObjectModel/ParseUtil.cpp]: (the shared model changes here are at least mostly from my highlight change that just went in, and won't show up once this is updated to the latest master)

---
In reply to: [271827206](https://github.com/Microsoft/AdaptiveCards/pull/2609#discussion_r271827206) [](ancestors = 271827206,271454369)

**Pull request merged**